### PR TITLE
bugfix [OSS-67]: update import of pinecone exception

### PR DESCRIPTION
**Pull request by potter-potter**
the pinecone python package moved their importing of PineconeApiException

Chroma `sleep` added because even thought there is a `wait`, there is still some sort of timing issue.

### CODE REVIEW

File: unstructured/ingest/connector/pinecone.py [hunk 1]
```diff
@@ -92,12 +92,12 @@ def check_connection(self):
     @DestinationConnectionError.wrap
     @requires_dependencies(["pinecone"], extras="pinecone")
     def upsert_batch(self, batch):
-        import pinecone.core.client.exceptions
+        import pinecone.exceptions
```
Comment (MthwRobinson) [inline unstructured/ingest/connector/pinecone.py:95]: Phew! Glad that was a simple fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,13 @@
-## 0.15.1-dev0
+## 0.15.1-dev1
 
 ### Enhancements
 
 ### Features
 
 ### Fixes
 
+* **Update import of Pinecone exception**
```
Comment (MthwRobinson) [inline CHANGELOG.md:9]: Could you add a note that this adds compatibility for `pinecone-client>=5.0.0`?

**Review by MthwRobinson — APPROVED** [review]
Could you also remove the `pinecone-client` pin in `constraint.txt` and do a fresh `make pip-compile`? Otherwise looks good!

**Comment by CJNA** [comment]
Hope this gets deployed to next release soon.
Meanwhile, I will use this branch

**Pull request merged**